lib/vrl/stdlib/src/encode_logfmt.rs

```diff
@@ -0,0 +1,227 @@
+use vrl::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeLogfmt;
+
+impl Function for EncodeLogfmt {
+    fn identifier(&self) -> &'static str {
+        "encode_logfmt"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::OBJECT | kind::ARRAY,
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+
+        Ok(Box::new(EncodeLogfmtFn { value }))
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "encode object",
+                source: r#"encode_logfmt({"lvl": "info", "msg": "This is a message", "log_id": 12345})"#,
```

Comment:
Does this handle nested objects and arrays? For example, I would expect:

```
{ "foo": { "bar": 1, "baz": [{"boop": 2}] }}
```

to encode as:

```
foo.bar=1 foo.baz.0.boop=2
```

Reply:
Currently, I implemented it using the `fmt::Display` trait implementation for `Value`.

I expect the result to be:

```
foo="{ "bar": 1, "baz": [{"boop": 2}] }"
```

Note that as you pointed out, quotes `"` are not properly escaped.

I don't think that nested keys are a common thing in `logfmt`.
What do you think ? Should I add a test-case ?

Reply:
👍 I think it would be nice to flatten them out. That's what I'm used to seeing for logfmt anyway.